### ISSUE TITLE
iOS - Chat - Suggestion list does not open when entering @ in personal or group chat

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#28657]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open iOS App
2. Login with any account
3. Open any personal or group chat
4. Type @

## Expected Result:
Suggestion list opens immediately

## Actual Result:
Nothing happens
Suggestion list opens if user delete @ and type it again

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.76-1

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/f56ae7af-3ba3-46fe-8a7c-4bab8edf6dc1

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by s-alves10** [comment]
This is a regression of [this PR](https://github.com/Expensify/App/pull/28583)
This issue is reproducible only in staging. It's a deploy blocker, I think

**Comment by amyevans** [comment]
Well we deployed about 8 hours ago, so now it is reproducible in production 😅. So no longer a deploy blocker, but still a regression that should be handled by the original participants from https://github.com/Expensify/App/pull/28583. @lakchote @cubuspl42 @paultsimura could you take a look here please? Thanks!

**Comment by cubuspl42** [comment]
This is bad.

@lanitochka17 Does this affect only iOS Native?

I created a PR reverting the original change: https://github.com/Expensify/App/pull/28794. I will create a thread on Slack.

**Comment by Julesssss** [comment]
We're going to revert the changes [here](https://github.com/Expensify/App/pull/28794).

**Comment by cubuspl42** [comment]
Confirmed reproducible on iOS Native:

https://github.com/Expensify/App/assets/2590174/10a94999-6a3e-49a7-8ffa-50b694a46d09



Confirmed **not reproducible** on Android Native:

https://github.com/Expensify/App/assets/2590174/f803035e-eff3-489d-aa6f-d7b5fd7749fe

**Comment by paultsimura** [comment]
I'm sorry guys, it's totally my bad.
I'm currently working on finding the root cause of why iOS native has an order of `selection` & `value` changes different from all other platforms.

**Comment by Julesssss** [comment]
Hey @cubuspl42 can you please fill the author checklist for the revert PR so I can review/merge/cp it?

**Comment by cubuspl42** [comment]
@paultsimura 

> I'm sorry guys, it's totally my bad.

It also passed through my review, the internal engineer approval, `main`, and QA tests on staging 🙂 We all sign under each change in a way.

**Comment by cubuspl42** [comment]
Not only is this inconsistent between platforms, but also between _type of a report_. We tested on iOS Native, after all...

**Comment by amyevans** [comment]
Yeah don't beat yourself up about it! The important thing is you've been active helping to fix it when a problem _is_ uncovered 😄

**Comment by cubuspl42** [comment]
@paultsimura Please get familiar with [this discussion on Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1696419713852579?thread_ts=1696409993.543749&cid=C01GTK53T8Q)

**Comment by shubham1206agra** [comment]
BTW there is a slack discussion going on right now
https://expensify.slack.com/archives/C01GTK53T8Q/p1696421170146759

**Comment by dylanexpensify** [comment]
@cubuspl42 mind giving an update here?

**Comment by paultsimura** [comment]
@dylanexpensify We are waiting for @situchan's response here in slack:

https://expensify.slack.com/archives/C01GTK53T8Q/p1696421170146759

**Comment by amyevans** [comment]
I completed the checklist above - I think we can close this out since the revert is complete and no payments are due. A new plan of action for addressing the original bug can continue on Slack / in https://github.com/Expensify/App/issues/28170.

If anyone disagrees though or if I missed something feel free to reopen!
